engine/hatchery/local/local.go

```diff
@@ -151,6 +164,47 @@ func (h *HatcheryLocal) Serve(ctx context.Context) error {
 	return h.CommonServe(ctx, h)
 }
 
+func (h *HatcheryLocal) downloadWorker() error {
+	urlBinary := h.Client.DownloadURLFromAPI("worker", sdk.GOOS, sdk.GOARCH, "")
+
+	log.Debug("downloading worker binary from %s", urlBinary)
+	resp, err := http.Get(urlBinary)
+	if err != nil {
+		return sdk.WrapError(err, "error while getting binary from CDS API")
+	}
+	defer resp.Body.Close()
+
+	if contentType := sdk.GetContentType(resp); contentType != "application/octet-stream" {
+		return fmt.Errorf("Invalid Binary (Content-Type: %s). Please try again or download it manually from %s", contentType, sdk.URLGithubReleases)
+	}
+
+	if resp.StatusCode != 200 {
+		return fmt.Errorf("Error http code: %d, url called: %s", resp.StatusCode, urlBinary)
+	}
+
+	workerFullPath := path.Join(h.BasedirDedicated, "worker")
+
+	if _, err := os.Stat(workerFullPath); err == nil {
+		log.Debug("removing existing worker binary from %s", workerFullPath)
+		if err := os.Remove(workerFullPath); err != nil {
+			return sdk.WrapError(err, "error while removing existing worker binary %s", workerFullPath)
+		}
+	}
+
+	log.Debug("copy worker binary into %s", workerFullPath)
+	fp, err := os.OpenFile(workerFullPath, os.O_CREATE|os.O_WRONLY|os.O_TRUNC, 0700)
+	if err != nil {
+		return sdk.WithStack(err)
+	}
+	defer fp.Close()
```

Comment:
you should check this error because it flushes the file - want it could hide a useful error

Reply:
done